source/appModules/eclipse.py

```diff
@@ -181,10 +181,15 @@ def chooseNVDAObjectOverlayClasses(self, obj, clsList):
 		try:
 
 			# Autocompletion items are placed outside the main eclipse window
-			if (obj.role == controlTypes.Role.LISTITEM
+			if (
+				obj.role == controlTypes.Role.LISTITEM
 				and obj.parent.parent.parent.role == controlTypes.Role.DIALOG
-				and obj.parent.parent.parent.parent.parent == api.getDesktopObject()
-				and obj.parent.parent.parent.parent.simpleNext.role == controlTypes.Role.BUTTON):
+				and obj.parent.parent.parent.simpleParent == api.getDesktopObject()

```

Comment:
Shouldn't the previous condition still remain? I'm concerned of the side effects of breaking the previous condition.

```suggestion
				and (
					obj.parent.parent.parent.simpleParent == api.getDesktopObject()
					or obj.parent.parent.parent.parent.parent == api.getDesktopObject()
				)
```

Reply:
I'll perform some tests on the Windows 11 machine with the suggested change, the reason I refactored it to the simple object navigation mode was to ease maintanability of the code, Windows 11 introduced a wrapping object below the desktop object and this broke the check, I'm also thinking if there is another way to refactor this check completely and not rely on the dialog's position on the tree, but I didn't find currently a solution that didn't involve the position on the tree... if Windows introduces another change, if the start menu becomes a checkbox or another control is placed right of the dialog for example, the code would break again...